extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiConfig.java

```diff
@@ -0,0 +1,267 @@
+package io.quarkus.swaggerui.deployment;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.openapi.ui.DocExpansion;
+import io.smallrye.openapi.ui.HttpMethod;
+import io.smallrye.openapi.ui.ThemeHref;
+
+@ConfigRoot
+public class SwaggerUiConfig {
+
+    /**
+     * The path where Swagger UI is available.
+     * <p>
+     * The value `/` is not allowed as it blocks the application from serving anything else.
+     */
+    @ConfigItem(defaultValue = "/swagger-ui")
+    String path;
+
+    /**
+     * If this should be included every time. By default this is only included when the application is running
+     * in dev mode.
+     */
+    @ConfigItem
+    boolean alwaysInclude;
+
+    /**
+     * If Swagger UI should be enabled. By default, Swagger UI is enabled.
+     */
+    @ConfigItem(defaultValue = "true")
+    boolean enable;
```

Comment:
Between `enable` and `deactivate`, it's going to be a bit hard to guess what will be going on :).

I wonder if we should simply move this one to runtime and have something like:
- alwaysInclude: defines if Swagger UI is included in the package or not
- enable: defines if it's exposed or not. And this would be a runtime config.

Reply:
I initially just move enable to a runtime config. However, then you can not disable in dev and test mode. So I decided to allow you to `deactivate`, if included. Maybe it should change to `deactivateIfIncluded` ?

Reply:
Except if we say you can not disable in dev and test ? And then use disable to disable in runtime when included ?

Reply:
> Between `enable` and `deactivate`, it's going to be a bit hard to guess what will be going on :).

This is my initial reaction to.
 
> I wonder if we should simply move this one to runtime and have something like:
> 
> * alwaysInclude: defines if Swagger UI is included in the package or not
> * enable: defines if it's exposed or not. And this would be a runtime config.

+1 on this. 

> I initially just move enable to a runtime config. However, then you can not disable in dev and test mode

Not sure I understood the issue but I think defining the `enable=false` property in the `application.properties` on a profile basis can fix it.

Reply:
Ok, so if we just move enable to runtime, and alwaysInclude stay as is that is fine? Should I rename enable to expose ?

Reply:
> Ok, so if we just move enable to runtime, and alwaysInclude stay as is that is fine?

Yes, exactly.

`alwaysInclude` will only decide if we include the bits outside of dev mode
`enable` will allow to enable things and will be runtime. Obviously, it will only enable things if the bits are included.

> Should I rename enable to expose ?

No, I would keep it as it is to avoid breaking compatibility for nothing. The `expose` idea was before I stumbled upon this `enable` knob.

As said by @machi1990 , I think profiles will allow to handle everything nicely.

Reply:
Ok, let me update, I'll ping you once everything is changed